core/src/main/java/io/questdb/griffin/engine/join/LtJoinNoKeyFastRecordCursorFactory.java

```diff
@@ -0,0 +1,362 @@
+            cursor.of(masterCursor, slaveCursor);
+            return cursor;
+        } catch (Throwable e) {
+            Misc.free(slaveCursor);
+            Misc.free(masterCursor);
+            throw e;
+        }
+    }
+
+    @Override
+    public int getScanDirection() {
+        return masterFactory.getScanDirection();
+    }
+
+    @Override
+    public boolean recordCursorSupportsRandomAccess() {
+        return false;
+    }
+
+    @Override
+    public void toPlan(PlanSink sink) {
+        sink.type("Lt Join Fast Scan");
+        sink.child(masterFactory);
+        sink.child(slaveFactory);
+    }
+
+    @Override
+    protected void _close() {
+        ((JoinRecordMetadata) getMetadata()).close();
+        masterFactory.close();
+        slaveFactory.close();
+    }
+
+    private static class AsOfJoinFastRecordCursor implements NoRandomAccessRecordCursor {
+        private final int columnSplit;
+        private final int lookahead;
+        private final int masterTimestampIndex;
+        private final OuterJoinRecord record;
+        private final int slaveTimestampIndex;
+        private boolean isMasterHasNextPending;
+        // stands for forward/backward scan through slave's time frames
+        private boolean isSlaveForwardScan;
+        private boolean isSlaveOpenPending;
+        private RecordCursor masterCursor;
+        private boolean masterHasNext;
+        private Record masterRecord;
+        private TimeFrameRecordCursor slaveCursor;
+        private int slaveFrameIndex = -1;
+        private long slaveFrameRow = Long.MIN_VALUE;
+        private Record slaveRecA;
+        private Record slaveRecB;
+        private long slaveTimestamp = Long.MIN_VALUE;
+
+        public AsOfJoinFastRecordCursor(
+                int columnSplit,
+                Record nullRecord,
+                int masterTimestampIndex,
+                int slaveTimestampIndex,
+                int lookahead
+        ) {
+            this.columnSplit = columnSplit;
+            this.record = new OuterJoinRecord(columnSplit, nullRecord);
+            this.masterTimestampIndex = masterTimestampIndex;
+            this.slaveTimestampIndex = slaveTimestampIndex;
+            this.lookahead = lookahead;
+        }
+
+        @Override
+        public void calculateSize(SqlExecutionCircuitBreaker circuitBreaker, Counter counter) {
+            masterCursor.calculateSize(circuitBreaker, counter);
+        }
+
+        @Override
+        public void close() {
+            masterCursor = Misc.free(masterCursor);
+            slaveCursor = Misc.free(slaveCursor);
+        }
+
+        @Override
+        public Record getRecord() {
+            return record;
+        }
+
+        @Override
+        public SymbolTable getSymbolTable(int columnIndex) {
+            if (columnIndex < columnSplit) {
+                return masterCursor.getSymbolTable(columnIndex);
+            }
+            return slaveCursor.getSymbolTable(columnIndex - columnSplit);
```

Comment:
nit: not covered by tests

Reply:
Added a test to cover this branch in 4c3b75023bb6ce4df54a4e1ba6c2a8488da0755a